R/framework_ebp.R

```diff
@@ -90,38 +106,41 @@ framework_ebp <- function(fixed, pop_data, pop_domains, smp_data, smp_domains,
   )
 
   indicator_list <- list(
-    fast_mean = function(y, threshold) {
-      t(mean(y))
+    fast_mean = function(y, w, threshold) {
+      t(weighted.mean(y, w))
     },
-    hcr = function(y, threshold) {
-      t(mean(y < threshold))
+    hcr = function(y, w, threshold) {
+       t(weighted.mean(y < threshold, w))
     },
-    pgap = function(y, threshold) {
-      t(mean((y < threshold) * (threshold - y) / threshold))
+    pgap = function(y, w, threshold) {
+      sum((1 - (y[y < threshold]/threshold)) * w[y < threshold])/sum(w)
     },
-    gini = function(y, threshold) {
-      n <- length(y)
-      y <- sort(y)
-      G <- sum(y * 1L:n)
-      G <- 2 * G / sum(y) - (n + 1L)
-      G <- t(G / n)
-      return(G)
+    gini = function(y, w, threshold) {
+        n <- length(y)
+        w <- w[order(y)]
```

Comment:
Are we sure that order and sort sort in the same way?

Reply:
Habe ich getestet und bei den Tests den Gini mit aufgenommen.